### PR TITLE
Create add_remove_local_repositories how to guide

### PR DESCRIPTION
**Pull request by david-sandy**
Created a how to guide for adding or removing repositories from the local development environment.

### CODE REVIEW

File: doc/dev/how-to/add_remove_local_repositories.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      "url": "https://ghe.sgdev.org",
+      "token": "1ec8973c1debb3232468b514b1a850dba87ee29a",
+      "repositoryQuery": ["org:sourcegraph gorilla"]
+    }
+  ],
+```
+
+**BEFORE** you make any changes, it is **strongly recommended** that you make a local backup of this file so that you can easily revert any changes if needed.
+
+To add or remove repositories, change the entires in the **repos** array:
```
Comment (kopancek) [inline doc/dev/how-to/add_remove_local_repositories.md:41]:
```suggestion
To add or remove repositories, change the entries in the **repos** array:
```

**Review by kopancek — APPROVED** [review]
LGTM. Added a suggestion with typo fix.

File: doc/dev/how-to/add_remove_local_repositories.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+Your local development environment must be restarted to pick up any changes you have made.
+
+Find this section of the code: 
+
+```
+  "GITHUB": [
+    {
+      "authorization": {},
+      "url": "https://github.com",
+      "token": "f189fe46ca1901b4988936cbc671f8305715ed73", // unprivileged sqs-test user
```
Comment (mrnugget) [inline doc/dev/how-to/add_remove_local_repositories.md:18]: Uhm. I don't think this should be in here?

File: doc/dev/how-to/add_remove_local_repositories.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        "sourcegraph-testing/etcd",
+        "sourcegraph-testing/tidb",
+        "sourcegraph-testing/titan",
+        "sourcegraph-testing/zap"
+      ],
+      "cloudDefault": true
+    },
+    {
+      "url": "https://ghe.sgdev.org",
+      "token": "1ec8973c1debb3232468b514b1a850dba87ee29a",
```
Comment (mrnugget) [inline doc/dev/how-to/add_remove_local_repositories.md:33]: Same?

File: doc/dev/how-to/add_remove_local_repositories.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
+# How to add or remove repositories from your local environment
+
+Depending on your needs, you may want to add or remove repositories from your local environment.  
+Fewer repositories will result in a quicker start time and less resources used on your local computer.
+
+The default repositories to load are located in a file named `external-services-config.json`.  
+This file is located here: `/dev-private/enterprise/dev/external-services-config.json` on your local development environment.
```
Comment (mrnugget) [inline doc/dev/how-to/add_remove_local_repositories.md:7]: I think this could be misleading, since `/dev-private` is a clone of the `dev-private` repository. I'd rephrase this to say something like "It's in the clone of `dev-private"

File: doc/dev/how-to/add_remove_local_repositories.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      "repos": [
+        "sourcegraph/sourcegraph",
+        "hashicorp/go-multierror",
+        "hashicorp/errwrap",
+        "sourcegraph-testing/etcd",
+        "sourcegraph-testing/tidb",
+        "sourcegraph-testing/titan",
+        "sourcegraph-testing/zap"
+      ],
+```
```
Comment (mrnugget) [inline doc/dev/how-to/add_remove_local_repositories.md:53]: I don't think it's worth it to showcase this array again, without any changes.

File: doc/dev/how-to/add_remove_local_repositories.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+## Removing a repository
+
+To remove one or more repositories from your local environment, simply remove those lines.  
+Remember to ensure a comma is present after every item in the list except the last item.
+
+For example, to remove the "sourcegraph-testing/zap" and "sourcegraph-testing/etcd" repositories:
+
+```
+```
```
Comment (mrnugget) [inline doc/dev/how-to/add_remove_local_repositories.md:85]: One line needs to be removed

File: doc/dev/how-to/add_remove_local_repositories.md (hunk 5 above)
Comment (mrnugget) [inline doc/dev/how-to/add_remove_local_repositories.md:53]: I don't think it's worth it to showcase this array again, without any changes.

File: doc/dev/how-to/add_remove_local_repositories.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    },
+    {
+      "url": "https://ghe.sgdev.org",
+      "token": "1ec8973c1debb3232468b514b1a850dba87ee29a",
+      "repositoryQuery": ["org:sourcegraph gorilla"]
+    }
+  ],
+```
+
+**BEFORE** you make any changes, it is **strongly recommended** that you make a local backup of this file so that you can easily revert any changes if needed.
```
Comment (mrnugget) [inline doc/dev/how-to/add_remove_local_repositories.md:39]: Do we need to call this out? It's in `git`, so they can always reclone the repository?

File: doc/dev/how-to/add_remove_local_repositories.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+Find this section of the code: 
+
+```
+  "GITHUB": [
+    {
+      "authorization": {},
+      "url": "https://github.com",
+      "token": "f189fe46ca1901b4988936cbc671f8305715ed73", // unprivileged sqs-test user
+      "repositoryQuery": ["affiliated"],
```
Comment (mrnugget) [inline doc/dev/how-to/add_remove_local_repositories.md:19]: This one here is the killer, btw. This is what causes a lot of clones. I'd call out this line.

**Review by mrnugget — CHANGES_REQUESTED** [review]
Please remove the tokens and rotate them in `dev-private` and let people know in #dev-announce.

File: doc/dev/how-to/add_remove_local_repositories.md (hunk 4 above)
Comment (david-sandy) [reply doc/dev/how-to/add_remove_local_repositories.md:7]: That's a good point!  We don't want any confusion.

File: doc/dev/how-to/add_remove_local_repositories.md (hunk 5 above)
Comment (david-sandy) [reply doc/dev/how-to/add_remove_local_repositories.md:53]: Gotcha!

File: doc/dev/how-to/add_remove_local_repositories.md (hunk 7 above)
Comment (david-sandy) [reply doc/dev/how-to/add_remove_local_repositories.md:39]: That is true, but, I was thinking of the scenario where they could have made additional changes before this one.  Is that a valid scenario?

File: doc/dev/how-to/add_remove_local_repositories.md (hunk 8 above)
Comment (david-sandy) [reply doc/dev/how-to/add_remove_local_repositories.md:19]: Do you mean "repositoryQuery": ["affiliated"]?

Comment (mrnugget) [reply doc/dev/how-to/add_remove_local_repositories.md:19]: Yep, see here: https://docs.sourcegraph.com/admin/external_service/github#configuration

Comment (david-sandy) [reply doc/dev/how-to/add_remove_local_repositories.md:19]: I see!  Great point!  Gold star for you!

File: doc/dev/how-to/add_remove_local_repositories.md (hunk 6 above)
Comment (david-sandy) [reply doc/dev/how-to/add_remove_local_repositories.md:85]: Check.

File: doc/dev/how-to/add_remove_local_repositories.md (hunk 8 above)
Comment (david-sandy) [reply doc/dev/how-to/add_remove_local_repositories.md:19]: I split the changes into two sections, one for the repos array and one for the repositoryQuery value.

**Comment by eseliger** [comment]
As part of #wg-delete-week I am going to close this PR. If it’s still needed, feel free to reopen and restore the branch.
